yt_dlp/extractor/adobepass.py

```diff
@@ -1574,18 +1574,34 @@ def extract_redirect_url(html, url=None, fatal=False):
                             post_form(mvpd_confirm_page_res, 'Confirming Login')
                 elif mso_id == 'Philo':
                     # Philo has very unique authentication method
-                    self._download_webpage(
-                        'https://idp.philo.com/auth/init/login_code', video_id, 'Requesting auth code', data=urlencode_postdata({
-                            'ident': username,
-                            'device': 'web',
-                            'send_confirm_link': False,
-                            'send_token': True,
-                        }))
+                    philo_ident_payload = {
+                        'ident': username,
+                        'device': 'web',
+                        'send_confirm_link': False,
+                        'send_token': True,
+                        'device_ident': f'web-{uuid.uuid4().hex}',
+                        'include_login_link': True,
+                    }
+
+                    self._download_json_handle(
+                        'https://idp.philo.com/auth/init/login_code', video_id,
+                        note='Requesting Philo auth code', data=json.dumps(philo_ident_payload).encode('utf-8'),
+                        headers={
+                            'Content-Type': 'application/json',
+                            'Accept': 'application/json',
+                        })
```

Comment:
- inline the payload since it's only used once
- `utf-8` is the default
- since we don't need to read the content of the response, just use `_request_webpage`
```suggestion
                    self._request_webpage(
                        'https://idp.philo.com/auth/init/login_code', video_id,
                        'Requesting Philo auth code', data=json.dumps({
                            'ident': username,
                            'device': 'web',
                            'send_confirm_link': False,
                            'send_token': True,
                            'device_ident': f'web-{uuid.uuid4().hex}',
                            'include_login_link': True,
                        }).encode(), headers={
                            'Content-Type': 'application/json',
                            'Accept': 'application/json',
                        })
```